x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -0,0 +1,600 @@
+Then success message should be displayed after upgrade
+And user should NOT see a CTA to upgrade prebuilt rules
+And user should NOT see a number of rules available to upgrade
+And user should NOT see the Rule Updates table
+```
+
+### Rule upgrade workflow: filtering, sorting, pagination
+
+TODO: add scenarios
+
+### Rule upgrade workflow: misc cases
+
+#### **Scenario: User doesn't see the Rule Updates tab until the package installation is completed**
+
+**Automation**: unit tests.
+
+```Gherkin
+Given prebuilt rules package is not installed
+When user opens the Rule Management page
+Then user should NOT see the Rule Updates tab until the package installation is completed and there are rules available for upgrade
+```
+
+### Error handling
+
+#### **Scenario: Error is handled when any operation on prebuilt rules fails**
+
+**Automation**: unit tests.
+
+```Gherkin
+When user is <operation> prebuilt rules
+And this operation fails
+Then user should see an error message
+
+Examples:
+  | operation             |
+  | installing all        |
+  | installing selected   |
+  | installing individual |
+  | upgrading all         |
+  | upgrading selected    |
+  | upgrading individual  |
+```
+
+### Authorization / RBAC
+
+#### **Scenario: User with read privileges on Security Solution cannot install prebuilt rules**
+
+**Automation**: 1 e2e test with mock rules + 3 integration tests with mock rules for the status and installation endpoints.
+
+```Gherkin
+Given user with "Security: read" privileges on Security Solution
+And no prebuilt rules are installed in Kibana
+And there are prebuilt rules available to install
+When user opens the Add Rules page
+Then user should see prebuilt rules available to install
+But user should not be able to install them
+```
+
+#### **Scenario: User with read privileges on Security Solution cannot upgrade prebuilt rules**
+
+**Automation**: 1 e2e test with mock rules + 3 integration tests with mock rules for the status and upgrade endpoints.
+
+```Gherkin
+Given user with "Security: read" privileges on Security Solution
+And X prebuilt rules are installed in Kibana
+And for Y of the installed rules there are new versions available
+When user opens the Rule Management page
+And user opens the Rule Updates table
+Then user should see prebuilt rules available to upgrade
+But user should not be able to upgrade them
+```
+
+### Kibana upgrade
+
+#### **Scenario: User can use prebuilt rules after upgrading Kibana from version A to B**
+
+**Automation**: not automated, manual testing required.
+
+```Gherkin
+Given user is upgrading Kibana from version <A> to version <B>
+And the <A> instance contains already installed prebuilt rules
+When the upgrade is complete
+Then user should be able to install new prebuilt rules
+And delete installed prebuilt rules
+And upgrade installed prebuilt rules that have newer versions in <B>
+
+Examples:
+  | A      | B     |
+  | 8.7    | 8.9.0 |
+  | 7.17.x | 8.9.0 |
+```
```

Comment:
@vgomez-el Because 8.9.0 is not the only target ("B") version that this scenario should work with. When 8.9.0 is released, the scenario will apply to 8.10.0, etc.